src/main/java/tempest/ui/MyCardLayout.java

```diff
@@ -0,0 +1,44 @@
+package tempest.ui;
+
+import tempest.ui.components.charts.ChartTypes;
+import tempest.ui.pages.PageNames;
+
+import java.awt.*;
+
+//Coded with help of: https://stackoverflow.com/questions/8277834/how-to-set-a-jframe-size-to-fit-the-cardlayout-displayed-jpanel
+
+public class MyCardLayout extends CardLayout {
+
+    @Override
+    public Dimension preferredLayoutSize(Container parent) {
+        Component current = findCurrentComponent(parent);
+
+        if (current.getName() != null) {
+            String name = current.getName();
+
+            //System.out.println("Name : " + name);
```

Comment:
I left these prints commented out as they'll be helpful for later if we do rework the charts